brax/jumpy.py

```diff
@@ -30,17 +33,21 @@
 float32 = onp.float32
 int32 = onp.int32
 
+FORCE_JNP = os.getenv("JUMPY_FORCE_JNP", 'False').lower() in ('true', '1', 't')
 
 def _in_jit() -> bool:
   """Returns true if currently inside a jax.jit call."""
+  if FORCE_JNP:
+    return True
   return core.cur_sublevel().level > 0
 
 
 def _which_np(*args):
-  """Returns np or jnp depending on args."""
-  for a in args:
-    if isinstance(a, jnp.ndarray) and not isinstance(a, onp.ndarray):
-      return jnp
+  checker = lambda a: (
+    isinstance(a, (jnp.ndarray, jax.interpreters.batching.BatchTracer))
+    and not isinstance(a, onp.ndarray))
+  if builtins.any(tree.flatten(tree.map_structure(checker, args))):
```

Comment:
can you use jax.tree_map here instead?  that way we can avoid importing `tree`

it's possible that brax already requires `tree` via some other dependency, but i'd prefer to avoid importing it, as then we'd have to add it to the setup.py requirements.

Reply:
Makes sense. I pushed a change that uses `jax.tree_map` and `jax.tree_flatten` for this.